server/modelpath.go

```diff
@@ -172,3 +173,33 @@ func GetBlobsPath(digest string) (string, error) {
 
 	return path, nil
 }
+
+func migrateRegistryDomain() error {
+	manifests, err := GetManifestPath()
+	if err != nil {
+		return err
+	}
+
+	olddomainpath := filepath.Join(manifests, "registry.ollama.ai")
+	newdomainpath := filepath.Join(manifests, DefaultRegistry)
+
+	return filepath.Walk(olddomainpath, func(path string, info fs.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if !info.IsDir() {
+			slog.Info("migrating registry domain", "path", path)
+			newpath := filepath.Join(newdomainpath, strings.TrimPrefix(path, olddomainpath))
+			if err := os.MkdirAll(filepath.Dir(newpath), 0o755); err != nil {
+				return err
+			}
+
+			if err := os.Rename(path, newpath); err != nil {
```

Comment:
would this error if I've pulled the same model from `ollama.com` and `registry.ollama.ai` ? there would be a collision there

Reply:
this will override the ollama.com manifest. that might not be the expected behaviour so more checks need to be added. however, the chances someone has an ollama.com manifest is extremely low